### DISCUSSION TITLE
Slaeh's CC2652RB stick not working

### DISCUSSION THREAD

**Discussion by mcividino** [Koenkk/zigbee2mqtt#5645] [Devices]
I have been trying to get this CR2652RB coordinator stick to work for several days with no luck.
I tried many different configuration changes, but zigbee2mqtt does not start at all.
I'm using IOTstack. This is what I did: 
   - stoped zigbee2mqtt docker container
   -  changed my configuration:
        serial --> ttyUSB0
        pan_id: 6758
   - plugged in my Saeh's stick
   - rebooted system
   - restarted the docker container

This is in the log file:
info  2021-01-09 11:28:43: Logging to console and directory: '/app/data/log/2021-01-09.11-28-41' filename: log.txt
info  2021-01-09 11:28:43: Starting Zigbee2MQTT version 1.17.0 (commit #07cdc9d)
info  2021-01-09 11:28:43: Starting zigbee-herdsman (0.13.46)
error 2021-01-09 11:29:52: Error while starting zigbee-herdsman

I was previously using a CC2531 USB dongle.
Did anyone tried with this stick? Is it possible that the stick wasn't properly flashed in origin?

Thanks.

**Comment by Koenkk** [comment]
I suggest relfashing first: https://slae.sh/projects/cc2652/#flashing, if that doesn't help provide the herdsman debug logging.

To enable herdsman debug logging, see https://www.zigbee2mqtt.io/information/debug.html#zigbee-herdsman-debug-logging

**Reply by mcividino** [reply]
I reflashed the units and tried all possible configuration changes. herdsman logging does not show any significant information.
I will assume the Slaeh's device just came defective and order the Electrolama one, which is what everybody seems to be using anyway.
Thanks for your help.

**Reply by Koenkk** [reply]
Looks the same as https://github.com/Koenkk/zigbee2mqtt/issues/5844 , can probably be fixed by powering the stick using a powered usb hub

**Reply by mcividino** [reply]
I will try it. I'm also using a Raspberry Pi 4 with an external SSD here.
Thanks

**Reply by mcividino** [reply]
I put it a powered USB hub and now it works.
Thanks for your help, sir.

**Reply by senarvi** [reply]
I got the same dongle from from slaesh and a Raspberry Pi 4 with an M.2 SSD. I first connected the dongle when the system was running. I logged in and noticed more and more files getting corrupted. I took the dongle out and a few restarts and file system checks later I saw that there was actually nothing wrong with the files on the disk. Maybe it was indeed a USB power problem. But I turned the power off and connected the dongle, and it's worked without problems since then.

**Reply by FlexxFR** [reply]
I have the same stick with a Raspberry Pi 4 and USB SSD, works fine. 

I had similar issue after a PI reboot, I was trying to reflash but I did not have the proper tools/time. I just put the stick back in and it worked! Last week it happened again (after some upgrades and cold pi-boot). Just put the stick in a different USB port, restart PI and it worked again. In the configuration I use the port name, not the number, so no re-config change needed.

serial:
  port: >-
    /dev/serial/by-id/usb-Silicon_Labs_slae.sh_cc2652rb_stick_-_slaesh_s_iot_stuff_00_12_4B_00_21_CB_F1_57-if00-port0

Don;t know why it sometime gets "stuck" on a certain USB, but it's an easy fix. Issue is rare, seems only happening after PI system/cold restart.  Could have something to do with powering the USB's. The SSD also has some issues sometimes are a restart.

**Comment by attilabalazsy** [comment]
Guys, thanks for your help! I had the same issue when I connected an SSD to my PI4. Before, it was using the microSD only, then I've connected and external SSD through a shield and moved the docker stuff there. Once I did my Slaeh's stick stopped working. Then I've reverted all my configs, but left the SSD connected. Zigbee was still not working. The log (even the low level herdsman log) was not showing any error, it was simply just not getting all the messages. What was strange, unpairing a device was always working, but the pairing and other communication not. Once I've disconnected the SSD shield, my zigbee was started to work again. When the SSD was connected back, the same issue was happening.
Now, I've just put the stick into another USB (previously I've tried this but not with all USBs) and it started to work also with connected SSD. No PI restart was needed. I'll do some more tests to see, if it is a permanent solution or just a lucky day :)

So, thanks again for the help here!

**Comment by mastier** [comment]
@Koenkk Let me tell you my story. I got exactly the same symptom like @attilabalazsy . 

My Supermicro embedded server got CPU issues and I needed to switch temporarily to AsRock N3150-ITX board and first time got this symptom. The slaesh's stick was able to receive all traffic from already paired devices, was even received network leave messages but was unable to send any command or pair new devices.
```
zigbee2mqtt    | [2024-11-22 10:37:33] info:    z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/0x847127fffec2e2f4', payload '{"battery":30,"humidity":29.5,"linkquality":131,"temperature":22.9,"voltage":2500}'
zigbee2mqtt    | [2024-11-22 10:37:34] info:    z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/0x847127fffec2e2f4', payload '{"battery":30,"humidity":29.5,"linkquality":134,"temperature":22.9,"voltage":2500}'
```
```
zigbee2mqtt    | [2024-11-22 07:15:25] error:   z2m: Publish 'set' 'state' to '0x54ef4410009ffe82' failed: 'Error: ZCL command 0x54ef4410009ffe82/1 genAnalogOutput.write({"presentValue":100}, {"timeout":10000,"disableResponse":false,"disa
bleRecovery":false,"disableDefaultResponse":true,"direction":0,"reservedBits":0,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205))'  
```
I tried flashing the newest firmware and same result:
https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/CC2652RB_coordinator_20240710.zip
Before that I got original from Slaesh when the stick was delivered while the Kickstarter/IndieGogo campaign.


I bought Raspberry Pi 5 + NVMe especially for that use case. And guess what...
SAME ISSUE :-D On purpose I bought DRAM-less so the NVMe is less power hungry (Crucial P3)



I used original 5V/5A power supply, before that (using 90W power delivery from my Dell display) I got quite often this message:
```
Nov 21 23:52:45 piatka kernel: cp210x ttyUSB0: failed set request 0x12 status: -110
```
With original power supply I saw this once. Is it really power issues or interference ?

I tried few times and once I succeeded with USB3.0 port but then it failed after few hours.
I also notices that when attached to powered hub it also works. I need check also with usb extension cord.
